src/Admin/AbstractAdmin.php

```diff
@@ -2023,9 +2035,35 @@ public function getChild($code)
         return $this->getChildren()[$code];
     }
 
-    public function setParent(AdminInterface $parent)
+    // NEXT_MAJOR: Uncomment the argument
+    public function setParent(AdminInterface $parent/*, string $parentAssociationMapping*/)
     {
         $this->parent = $parent;
+
+        // NEXT_MAJOR: Uncomment the following line
+        // $this->parentAssociationMapping[$parent->getCode()] = $parentAssociationMapping;
+
+        // NEXT_MAJOR: Remove next line.
+        $args = \func_get_args();
+
+        // NEXT_MAJOR: Remove this entire if-else block.
+        if (isset($args[1])) {
+            if (!\is_string($args[1])) {
+                throw new \TypeError(sprintf(
+                    'Argument 2 passed to "%s()" must be of the type string, "%s" given.',
+                    __METHOD__,
+                    \is_object($args[1]) ? \get_class($args[1]) : \gettype($args[1])
+                ));
+            }
+
+            $this->parentAssociationMapping[$parent->getCode()] = $args[1];
+        } else {
+            @trigger_error(sprintf(
+                'Not passing argument 2 to "%s()" is deprecated since'
```

Comment:
Passing `null` is different than "not passing". Please, note that the related `if` is using `isset()`.
IMO, we should count the provided arguments in order to determine if the argument was passed or not.